### PR TITLE
[docs] add metrics overview

### PR DESCRIPTION
**Pull request by polarweasel**
@netlify /preview/launch-and-manage/monitor-and-alert/metrics-overview/

### CODE REVIEW

File: docs/content/preview/explore/observability/metrics-overview.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+| pg_stat_activity | Per Node <br/> Current normalized view of queries running on the node. <br/> Equivalent view on YBA/YBM is _Live Queries_. | :13000/rpcz |
+
+## Metric syntax
+
+There are 4 major metric categories:
+
+* **Server** (logical architectural components) metrics per node
+* **Table** metrics per node
+* **Tablet** metrics per node
+* **Cluster** metrics
```
Comment (dshah-yugabyte) [inline docs/content/preview/explore/observability/metrics-overview.md:44]: Cluster metrics per Master

File: docs/content/preview/explore/observability/metrics-overview.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+Each node in a YugabyteDB cluster exports metrics through JSON and Prometheus Exposition Format API endpoint. You can aggregate the metrics from each node to get a cluster-wide view. YugabyteDB Anywhere (YBA) and YugabyteDB Managed (YBM) aggregate a sub-set of relevant metrics from each node and provide an aggregated cluster view.
+
+### Export by logical architecture components
+
+| Component | Description | JSON API endpoints | Prometheus API endpoint |
+| :-------- | :---------- | :----------------- | :---------------------------------------- |
+| YSQL | YSQL statement and connection metrics. <br/> Starting point for all YSQL query-related heuristics. | | :13000/metrics | :13000/prometheus-metrics |
+| YCQL | YCQL statement and connection metrics. <br/> Starting point for all YCQL query-related heuristics. | :12000/metrics | :12000/prometheus-metrics |
+| Master | | :7000/metrics | :7000/threadz | :7000/prometheus-metrics |
```
Comment (dshah-yugabyte) [inline docs/content/preview/explore/observability/metrics-overview.md:26]: Description:
Yugabyte Master Service, Postgres and storage layer metrics

Value:
Debug level metrics to troubleshoot system issues

File: docs/content/preview/explore/observability/metrics-overview.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+| TServer | | :9000/metrics | :9000/threadz | :9000/prometheus-metrics |
```
Comment (dshah-yugabyte) [inline docs/content/preview/explore/observability/metrics-overview.md:27]: Description:
Yugabyte Tablet Server service, Postgres and storage layer metrics

Value:
Debug level metrics to troubleshoot system and  storage issues

File: docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+---
+
+YugabyteDB provides nearly two thousand metrics for monitoring, performance-tuning, and troubleshooting system, table, and tablet issues.
+
+This document focuses primarily on metrics and their uses detailing about which metrics are the most important ones, how to consume the metrics in practice, and how to use them to monitor and manage clusters, troubleshoot performance issues, and identify bottlenecks.
+
+{{< note title="Note" >}}
+
+- This page does not contain an exhaustive list of all the metrics exported by YugabyteDB.
+- This page only covers metrics and not query tuning. To learn more about query tuning in YugabyteDB, please visit this page.
```
Comment (ddhodge) [inline docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md:22]: Link?

Comment (ddhodge) [inline docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md:22]:
```suggestion
- This page only covers metrics and not query tuning. To learn about query tuning in YugabyteDB, refer to Query tuning.
```

File: docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md [hunk 5]
```diff
@@ -0,0 +1,319 @@
...
+
+## Overview
+
+A YugabyteDB cluster comprises multiple nodes and services, each emitting metrics. These metrics are applicable at different levels: cluster-wide, per-node, per-table, and per-tablet. These metrics are exported through various endpoints in JSON, HTML, and Prometheus formats. You can aggregate these metrics appropriately to get a cluster-wide, per-database, per-table, or per-node view. The following diagram shows the various endpoints in a cluster.
+
+<!-- ### Logical architecture components -->
+
+The following table includes a brief description of the type of metrics exposed by each endpoint and the URL from which their metrics can be exported.
+
+| End-point name | Description | JSON format | Prometheus format |
```
Comment (ddhodge) [inline docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md:34]:
```suggestion
| Endpoint | Description | JSON format | Prometheus format |
```

File: docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md [hunk 6]
```diff
@@ -0,0 +1,319 @@
...
+| :------------ | :---------- | :---------- | :---------------------------------------- |
+| [YB-Master](../../../architecture/concepts/yb-master/) | YB-Master exposes metrics related to the system catalog, cluster-wide metadata (such as the number of tablets and tables), and cluster-wide operations (table creations/drops, and so on.). | `<node-ip>:7000/metrics` | `<node-ip>:7000/prometheus-metrics` |
```
Comment (ddhodge) [inline docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md:36]:
```suggestion
| [YB-Master](../../../architecture/concepts/yb-master/) | Exposes metrics related to the system catalog, cluster-wide metadata (such as the number of tablets and tables), and cluster-wide operations (table creations/drops, and so on). | `<node-ip>:7000/metrics` | `<node-ip>:7000/prometheus-metrics` |
```

File: docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md [hunk 7]
```diff
@@ -0,0 +1,319 @@
...
+| [YB-TServer](../../../architecture/concepts/yb-tserver/) | YB-Tserver exposes metrics related to end-user DML requests (such as table insert), which include tables, tablets, and storage-level metrics (such as Write-Ahead-Logging, and so on.). | `<node-ip>:9000/metrics` | `<node-ip>:9000/prometheus-metrics` |
```
Comment (ddhodge) [inline docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md:37]:
```suggestion
| [YB-TServer](../../../architecture/concepts/yb-tserver/) | Exposes metrics related to end-user DML requests (such as table insert), which include tables, tablets, and storage-level metrics (such as Write-Ahead-Logging, and so on). | `<node-ip>:9000/metrics` | `<node-ip>:9000/prometheus-metrics` |
```

File: docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md [hunk 8]
```diff
@@ -0,0 +1,319 @@
...
+| [YSQL](../../../api/ysql/) | This endpoint exposes the YSQL query processing and connection metrics, such as throughput and latencies for the various operations. | `<node-ip>:13000/metrics` | `<node-ip>:13000/prometheus-metrics` |
```
Comment (ddhodge) [inline docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md:38]:
```suggestion
| [YSQL](../../../api/ysql/) | Exposes the YSQL query processing and connection metrics, such as throughput and latencies for the various operations. | `<node-ip>:13000/metrics` | `<node-ip>:13000/prometheus-metrics` |
```

File: docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md [hunk 9]
```diff
@@ -0,0 +1,319 @@
...
+| [YCQL](../../../api/ycql/) | This endpoint exposes the YCQL query processing and connection metrics (such as throughput and latencies for various operations). | `<node-ip>:12000/metrics` | `<node-ip>:12000/prometheus-metrics` |
```
Comment (ddhodge) [inline docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md:39]:
```suggestion
| [YCQL](../../../api/ycql/) | Exposes the YCQL query processing and connection metrics (such as throughput and latencies for various operations). | `<node-ip>:12000/metrics` | `<node-ip>:12000/prometheus-metrics` |
```

File: docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md [hunk 10]
```diff
@@ -0,0 +1,319 @@
...
+
+### Database IOPS (reads and writes)
+
+The [YB-TServer](../../../architecture/concepts/yb-tserver/) is responsible for the actual I/O of client requests in a YugabyteDB cluster. Each node in the cluster has a YB-TServer, and each hosts one or more tablet peers.
+
+A description of key metrics in this category is listed in the following table:
+
+| Metrics | Unit | Type | Description |
+| :------ | :--- | :--- | :---------- |
+| `handler_latency_yb_tserver_TabletServerService_Read` | microseconds | counter | The time in microseconds to perform WRITE operations at a tablet level |
```
Comment (ddhodge) [inline docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md:126]:
```suggestion
| `handler_latency_yb_tserver_TabletServerService_Read` | microseconds | counter | The time in microseconds to perform READ operations at a tablet level |
```

File: docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md [hunk 11]
```diff
@@ -0,0 +1,319 @@
...
+
+The replication lag metrics are computed at a tablet level as the difference between Hybrid Logical Clock (HLC) time on the source's tablet server, and the hybrid clock timestamp of the latest record pulled from the source.
+
+A description of key metrics in this category is listed in the following table:
+
+| Metrics | Unit | Type | Description |
+| :------ | :--- | :--- | :---------- |
+| `async_replication_committed_lag_micros` | microseconds | counter | The time in microseconds for the replication lag on the target cluster. This metric is available only on the source cluster. |
+| `time_since_last_getchanges` | microseconds | counter | The time elapsed in microseconds from when the source cluster got a request to replicate from the target cluster. This metric is available only on the source cluster. |
+| `consumer_safe_time_lag` | | | The time elapsed in microseconds between the physical time and safe time. Safe time is when data has been replicated to all the tablets on the consumer cluster. This metric is available only on the target cluster. |
```
Comment (ddhodge) [inline docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md]: unit and type missing from 2 rows

File: docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md [hunk 12]
```diff
@@ -0,0 +1,319 @@
...
+
+A description of throughput and latency metrics for the storage (RocksDB) layer is listed in the following table:
+
+| Metrics | Unit | Type | Description |
+| :------ | :--- | :--- | :---------- |
+| `Rocksdb_number_db_next`  | keys | counter | Whenever a tuple is read/updated from the database, a request is made to RocksDB key. Each database operation will have multiple requests to RocksDB. The number of NEXT operations performed to lookup a key by RocksDB when a tuple is read/updated by the database. |
+| `Rocksdb_number_db_prev`  | keys | counter | The number of PREV operations performed to lookup a key by RocksDB when a tuple is read/updated from the database. |
+| `Rocksdb_number_db_seek`  | keys | counter | The number of SEEK operations performed to lookup a key by the RocksDB when a tuple is read/updated from the database. |
+| `Rocksdb_db_write_micros` | microseconds | counter | The time spent by RocksDB in microseconds to write data |
```
Comment (ddhodge) [inline docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md:176]: missing periods 3 rows

File: docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md [hunk 13]
```diff
@@ -0,0 +1,319 @@
...
+
+### Block cache
+
+When the data requested from YSQL layer is sitting in an SST File, it will be cached in RocksDb Block Cache. This is the fundamental cache that sits in RocksDB instead of YSQL layer. A block requires multiple touches before it is added to the multi-touch (hot) portion of the cache.
+
+A description of key metrics in this category is listed in the following table:
+
+| Metrics | Unit | Type | Description |
+| :------ | :--- | :--- | :---------- |
+| `rocksdb_block_cache_hit` | blocks | counter | The total number of block cache hits (cache index + cache filter + cache data) |
```
Comment (ddhodge) [inline docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md:190]: missing periods

File: docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md [hunk 14]
```diff
@@ -0,0 +1,319 @@
...
+| `rocksdb_block_cache_miss` | blocks | counter | The total number of block cache misses (cache index + cache filter + cache data) |
+| `block_cache_single_touch_usage` | blocks | counter | Blocks of data cached and read once by the YSQL layer are classified in single touch portion of the cache. The size (in bytes) of the cache usage by blocks having a single touch. |
+| `block_cache_multi_touch_usage` | blocks | counter | Blocks of data cached and read once by the YSQL layer are classified in the multi-touch portion of the cache. The size (in bytes) of the cache usage by blocks having multiple touches. |
```
Comment (ddhodge) [inline docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md:193]: "read once"? this is multi touch

File: docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md [hunk 15]
```diff
@@ -0,0 +1,319 @@
...
+
+### SST files
+
+RocksDB LSM-trees buffer incoming data in a memory buffer that, when full, is sorted, and flushed to disk in the form of a sorted run. When a sorted run is flushed to disk, it may be iteratively merged with existing runs of the same size. Overall, as a result of such iterative merges, the sorted runs on disk, also termed Sorted-String Table or SST files, form a collection of levels of exponentially increasing size with potentially overlapping key ranges across the levels.
+
+A description of key metrics in this category is listed in the following table:
+
+| Metrics | Unit | Type | Description |
+| :------ | :--- | :--- | :---------- |
+| `rocksdb_current_version_sst_files_size` | bytes | counter | The aggregate size of all SST files |
```
Comment (ddhodge) [inline docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md:218]: missing periods

File: docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md [hunk 16]
```diff
@@ -0,0 +1,319 @@
...
+
+Memtable is the first level of data storage where data is stored when you start inserting. It provides statistics about reading documents, which are essentially columns in the table. If a memtable is full, the existing memtable is made immutable and stored on disk as an SST file.
+
+Memtable has statistics about reading documents, which essentially are columns in the table.
+
+A description of key metrics in this category is listed in the following table:
+
+| Metrics | Unit | Type | Description |
+| :------ | :--- | :--- | :---------- |
+| `rocksdb_memtable_compaction_micros` | microseconds | counter | The total time in microseconds to compact an set of SST files. |
```
Comment (ddhodge) [inline docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md:246]:
```suggestion
| `rocksdb_memtable_compaction_micros` | microseconds | counter | The total time in microseconds to compact a set of SST files. |
```

File: docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md [hunk 17]
```diff
@@ -0,0 +1,319 @@
...
+
+These metrics are available per tablet and can be aggregated across the entire cluster using appropriate aggregations.
+
+### Write-Ahead-Logging (WAL)
+
+The Write Ahead Log (or WAL) is used to write and persist updates to disk on each tablet. The following table includes metrics which allow observing the performance of the WAL component:
+
+| Metrics | Unit | Type | Description |
+| :------ | :--- | :--- | :---------- |
+| `log_sync_latency` | microseconds | counter | The number of microseconds spent to fsync the WAL entries to disk. |
```
Comment (ddhodge) [inline docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md:258]: fsync?

File: docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md [hunk 18]
```diff
@@ -0,0 +1,319 @@
...
+
+## YB-Master metrics
+
+The [YB-Master](../../../architecture/concepts/yb-master/) hosts system metadata, records about tables in the system and locations of their tablets, users, roles, permissions, and so on. YB-Masters are also responsible for coordinating background operations such as schema changes, handling addition and removal of nodes from the cluster, automatic re-replication of data on permanent failures, and so on.
+
+A description of key metrics in this category is listed in the following table:
+
+| Metrics | Unit | Type | Description |
+| :------ | :--- | :--- | :---------- |
+| `handler_latency_yb_master_MasterClient_GetTabletLocations` | microseconds | counter | The number of microseconds spent on fetching the replicas from the master servers. This metric includes the number of times the locations of the replicas from the master server. |
```
Comment (ddhodge) [inline docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md:274]: the number of times the locations of the replicas are what from the master server?

File: docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md [hunk 19]
```diff
@@ -0,0 +1,319 @@
...
+| `handler_latency_yb_tserver_TabletServerService_Read` | microseconds | counter | The time in microseconds to reads the PostgreSQL system tables (during DDL). This metric includes the count or number of writes. |
```
Comment (ddhodge) [inline docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md:275]: to perform reads?
the count or number of reads.

File: docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md [hunk 20]
```diff
@@ -0,0 +1,319 @@
...
+| `handler_latency_yb_tserver_TabletServerService_Write` | microseconds | counter | The time in microseconds to reads the PostgreSQL system tables (during DDL). This metric includes the count or number of writes. |
```
Comment (ddhodge) [inline docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md:276]: to perform writes?

File: docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md [hunk 21]
```diff
@@ -0,0 +1,319 @@
...
+| `handler_latency_yb_master_MasterDdl_CreateTable` | microseconds | counter | The time in microseconds to create a table (during DDL). This metric includes the count of create table operations.|
+| `handler_latency_yb_master_MasterDdl_DeleteTable` | microseconds | counter | The time in microseconds to delete a table (during DDL). This metric includes the count of delete table operations.|
+
+A description of key metrics in this category is listed in the following table:
```
Comment (ddhodge) [inline docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md:280]: what's this? should this be a "These metrics are available"-type blurb?

File: docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md [hunk 22]
```diff
@@ -0,0 +1,319 @@
...
+
+| Metrics | Unit | Type | Description |
+| :------ | :--- | :--- | :---------- |
+| `handler_latency_yb_consensus_ConsensusService_UpdateConsensus` | microseconds | counter | The time in microseconds to replicate a batch of log entries from the leader to the follower. This metric includes the total count of the RPC method being invoked. |
+| `handler_latency_yb_consensus_ConsensusService_RequestConsensusVotes` | microseconds | counter | The time in microseconds by candidates to gather votes. This metric includes the total count of the RPC method being invoked. |
+| `handler_latency_yb_consensus_ConsensusService_ChangeConfig` | microseconds | counter | The time in microseconds by candidates to add or remove a peer from the Raft group. This metric includes the total count of the RPC method being invoked. |
+
+The throughput (Ops/Sec) can be calculated and aggregated for nodes across the entire cluster using appropriate aggregations
```
Comment (ddhodge) [inline docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md:297]:
```suggestion
The throughput (Ops/Sec) can be calculated and aggregated for nodes across the entire cluster using appropriate aggregations.
```

**Review by ddhodge — APPROVED** [review]

File: docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md [hunk 23]
```diff
@@ -25,9 +25,9 @@ This document focuses primarily on metrics and their uses detailing about which
 
 ## Overview
 
-A YugabyteDB cluster comprises multiple nodes and services, each emitting metrics. These metrics are applicable at different levels: cluster-wide, per-node, per-table, and per-tablet. These metrics are exported through various endpoints in JSON, HTML, and Prometheus formats. You can aggregate these metrics appropriately to get a cluster-wide, per-database, per-table, or per-node view. The following diagram shows the various endpoints in a cluster.
+A YugabyteDB cluster comprises multiple nodes and services, each emitting metrics. These metrics are applicable at different levels: cluster-wide, per-node, per-table, and per-tablet. The metrics are exported through various endpoints in JSON, HTML, and Prometheus formats. You can aggregate metrics appropriately to get a cluster-wide, per-database, per-table, or per-node view. The following diagram shows the various endpoints in a cluster.
```
Comment (ddhodge) [inline docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md:28]:
```suggestion
A YugabyteDB cluster comprises multiple nodes and services, each emitting metrics. Metrics can be for an entire cluster, or for a specific node, table, or tablet, and they can be aggregated into cluster, node, database, and table views. Metrics are exported through various endpoints in JSON, HTML, and Prometheus formats, as shown in the following diagram.
```

File: docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md [hunk 24]
```diff
@@ -14,7 +14,7 @@ type: docs
 
 YugabyteDB provides nearly two thousand metrics for monitoring, performance-tuning, and troubleshooting system, table, and tablet issues.
 
-This document focuses primarily on metrics and their uses detailing about which metrics are the most important ones, how to consume the metrics in practice, and how to use them to monitor and manage clusters, troubleshoot performance issues, and identify bottlenecks.
+The following sections focus primarily on metrics and their uses, including how to consume the metrics in practice, and how to use them to monitor and manage clusters, troubleshoot performance issues, and identify bottlenecks.
```
Comment (ddhodge) [inline docs/content/preview/launch-and-manage/monitor-and-alert/metrics-overview.md:17]:
```suggestion
The following sections describe how you can consume YugabyteDB metrics, and use them to monitor and manage clusters, troubleshoot performance issues, and identify bottlenecks.
```

**Review by ddhodge — APPROVED** [review]

**Pull request merged**